src/index/fetcher.rs

```diff
@@ -65,18 +65,34 @@ impl Fetcher {
     }
 
     let body = Value::Array(reqs).to_string();
-    let req = Request::builder()
-      .method(Method::POST)
-      .uri(&self.url)
-      .header(hyper::header::AUTHORIZATION, &self.auth)
-      .header(hyper::header::CONTENT_TYPE, "application/json")
-      .body(Body::from(body))?;
 
-    let response = self.client.request(req).await?;
-
-    let buf = hyper::body::to_bytes(response).await?;
-
-    let mut results: Vec<JsonResponse<String>> = serde_json::from_slice(&buf)?;
+    let mut results: Vec<JsonResponse<String>>;
+    let mut tries = 0;
+
+    loop {
+      results = match self.try_get_transactions(body.clone()).await {
+        Ok(results) => results,
+        Err(error) => {
+          if tries >= 5 {
+            return Err(anyhow!(
+              "Failed to fetch raw transactions after 5 retries. Error: {}",
+              error
+            ));
+          }
+
+          log::info!(
+            "Failed to fetch raw transactions, retrying. Error: {}",
+            error
+          );
+          tries += 1;
+
+          let sleep_time = tokio::time::Duration::from_millis(100 * u64::pow(tries, 2));
```

Comment:
```suggestion
          let sleep_time = tokio::time::Duration::from_millis(100 * u64::pow(2, retries));
```
I think we want this for exponential backoff.